dump/test/util/dump/DumpTest.java

```diff
@@ -372,7 +373,16 @@ public void testVersionUpdate() throws Exception {
          n++;
       }
       assertThat(n).as("Dump was not deleted after version upgrade").isEqualTo(0);
+      v4Dump.add(new BeanVersion4(1));
       v4Dump.close();
+
+      Dump<BeanVersion5> v5Dump = new Dump<>(BeanVersion5.class, dumpFile);
+      n = 0;
+      for ( BeanVersion5 b : v5Dump ) {
+         n++;
```

Comment:
We should check here that pruning actually happened, no?

Reply:
Test and version check code has been updated to ensure that dump version is overwritten with code version, iff the dump has been deleted / renamed / rewritten.